symfit/core/fit_results.py

```diff
@@ -147,4 +191,73 @@ def _array_safe_dict_eq(one_dict, other_dict):
         else: return True
 
     def __eq__(self, other):
-        return FitResults._array_safe_dict_eq(self.__dict__, other.__dict__)
\ No newline at end of file
+        return FitResults._array_safe_dict_eq(self.__dict__, other.__dict__)
+
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        state['minimizer'] = (type(state['minimizer']),
+                              state['minimizer'].objective,
+                              state['minimizer'].parameters)
+        return state
+
+    def __setstate__(self, state):
+        min_class, objective, parameters = state['minimizer']
+        state['minimizer'] = min_class(objective, parameters)
+        self.__dict__.update(state)
+
+    def _gof_qualifiers(self):
+        """
+        Based on the objective used, we can infer certain goodness of fit
+        (g.o.f.) qualifiers.
+
+        The ``objective_value`` itself always exists, and then depending on the
+        objective we also get the following:
+
+        - The coefficient of determination :math:`R^2` and :math:`\\chi^2` for
+          :class:`~symfit.core.objectives.LeastSquares` and
+          :class:`~symfit.core.objectives.VectorLeastSquares`.
+        - Likelihood and log-likelihood for
+          :class:`~symfit.core.objectives.LogLikelihood`.
```

Comment:
Can't you always calculate all? :)

Reply:
No I think that's a bit dangerous since they don't all have meaning at the same time. So I'd rather err on the side of caution, then risk having someone cite a number that doesn't actually exist for their problem. If they disagree for their particular problem they can always go ahead and calculate it manually :).

Reply:
Works for me. Future PR would be to expose those functions somewhere.